python/mocpy/fmoc/fmoc.py

```diff
@@ -0,0 +1,338 @@
+        Create a FrequencyMOC from a set of FrequencyMOC ranges at order 59.
+
+        Parameters
+        ----------
+        order : int, The resolution of the FrequencyMOC
+        ranges: `~numpy.ndarray`
+                 a N x 2 numpy array representing the set of depth 61 ranges.
+
+        Returns
+        -------
+        moc : `~mocpy.tmoc.TimeMOC`
+            The MOC
+        """
+        ranges = np.zeros((0, 2), dtype=np.uint64) if ranges is None else ranges
+
+        if ranges.shape[1] != 2:
+            raise ValueError(
+                f"Expected a N x 2 numpy ndarray but second dimension is {ranges.shape[1]}",
+            )
+
+        if ranges.dtype is not np.uint64:
+            ranges = ranges.astype(np.uint64)
+
+        index = mocpy.from_fmoc_ranges_array2(np.uint8(order), ranges)
+        return cls(cls.__create_key, index)
+
+    @classmethod
+    def from_frequencies(cls, order, frequencies):
+        """
+        Create a FrequencyMOC from a `astropy.units.Quantity` that are internally converted in `Hz`.
+
+        Parameters
+        ----------
+        order : int, The resolution of the FrequencyMOC: see `relative_precision_to_order`
+        frequencies : `astropy.units.Quantity`
+            Quantity converted internally in `Hz`
+
+        Returns
+        -------
+        frequency_moc : `~mocpy.fmoc.FrequencyMOC`
+        """
+        frequencies = frequencies.to(u.Hz)
+        frequencies = np.atleast_1d(frequencies)
+
+        store_index = mocpy.from_freq_values(order, frequencies)
+        return cls(cls.__create_key, store_index)
+
+    @classmethod
+    def from_frequency_ranges(cls, order, min_freq, max_freq):
+        """
+        Create a FrequencyMOC from a range defined by two `astropy.units.Quantity` converted in `Hz`.
+
+        Parameters
+        ----------
+        order : int, The resolution of the FrequencyMOC: see `relative_precision_to_order`
+        min_freq : `astropy.units.Quantity`
+            astropy quantity converted in Hz and defining the left part of the intervals
+        max_freq : `astropy.units.Quantity`
+            astropy quantity converted in Hz and defining the right part of the intervals
+
+        Returns
+        -------
+        frequency_moc : `~mocpy.fmoc.FrequencyMOC`
+        """
+        min_freq = min_freq.to(u.Hz)
+        min_freq = np.atleast_1d(min_freq)
+
+        max_freq = max_freq.to(u.Hz)
+        max_freq = np.atleast_1d(max_freq)
+
+        if min_freq.shape != max_freq.shape:
+            raise ValueError(
+                f"Mismatch between min_freq and max_freq of shapes {min_freq.shape} and {max_freq.shape}",
+            )
+
+        store_index = mocpy.from_freq_ranges(
+            order,
+            min_freq,
+            max_freq,
+        )
+        return cls(cls.__create_key, store_index)
+
+    @property
+    def min_freq(self):
+        """
+        Get the `~astropy.units.Quantity` frequency of the fmoc first observation.
```

Comment:
I don't understand 'first observation'. Should we not say somethink like 'minimal observed frequency'? Or is it something else?